### DISCUSSION TITLE
Is it safe to call `tokio::sync::mpsc::Receiver::try_recv()` in sync code?

### DISCUSSION THREAD

**Discussion by goofacz** [tokio-rs/tokio#6136] [Q&A] [answered]
I'm bringing async & sync code in one of my projects. Async code acts as producer and sync code acts as consumer. The consumer code is run in spawned OS thread and received work via `tokio::sync::mpsc::Receiver`.

Is it safe to call `tokio::sync::mpsc::Receiver::try_recv()` in sync code? Unlike for `blocking_recv`, it's not clearly sated in docs whether synchronous code can call `try_recv()`.

I get it's not declared as `async` function but I wonder whether there are any other - not obvious - reasons.

**Comment by hawkw** [comment, marked as answer]
Yes. The `try_recv()` method is not `async` and does not take a `task::Context` parameter, so it may be called outside of an async  context and will work correctly.
